src/main/java/duke/command/AddCommand.java

```diff
@@ -0,0 +1,189 @@
+     * @return The task to be added into the task list.
+     * @throws DukeException If task string does not contain dates (for deadline and event tasks),
+     * or has wrong date/time formatting.
+     */
+    public Task processTask(String delimiter, String taskType) throws DukeException {
+        //Task Name Only
+        String taskName = Arrays.stream(getArray()).takeWhile(e -> !e.equals(delimiter))
+                .skip(1).collect(Collectors.joining(" "));
+        //Date + Time, each in a single array cell
+        String[] dateTime = Arrays.stream(getArray()).dropWhile(e -> !e.equals(delimiter))
+                .skip(1).collect(Collectors.joining(" ")).split(" ");
+
+        if (dateTime.length < 1 || dateTime[0].equals("")) {
+            throw new DukeException("All deadline/event tasks must come with a date in yyyy-mm-dd format!");
+        }
+
+        //Array to store 3 fields - task name, date and time (if available)
+        ArrayList<String> nameDateTimeStrings = new ArrayList<>();
+        nameDateTimeStrings.add(taskName);
+
+        //More than necessary words or date and time in wrong format
+        if (dateTime.length > 2) {
+            throw new DukeException("Please make sure date is imputed in yyyy-mm-dd format. Any optional time"
+                    + " parameter should be in HHmm format. Don't add any more characters after the date and time!");
+        }
+        //Append date and time into newArray if they exist
+        for (int i = 0; i < dateTime.length; i++) {
+            if (!dateTime[i].equals("")) {
+                nameDateTimeStrings.add(dateTime[i]);
+            }
+        }
+
+        try {
+            LocalDate taskDate = LocalDate.parse(nameDateTimeStrings.get(1), DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+            //Make sure deadline set is in the future
+            LocalDate todayDate = LocalDate.now();
+            if (taskDate.isBefore(todayDate)) {
+                throw new DukeException("Date for deadline/event tasks must be set in the future!");
+            }
+            //Time Exists
+            if (nameDateTimeStrings.size() > 2) {
+                LocalTime taskTime = LocalTime.parse(nameDateTimeStrings.get(2), DateTimeFormatter.ofPattern("HHmm"));
+                //Make sure time has not passed
+                LocalTime timeNow = LocalTime.now();
+                if (todayDate.isEqual(taskDate) && taskTime.isBefore(timeNow)) {
+                    throw new DukeException("The date/time combination you specified has already passed!");
+                }
+                return (taskType.equals("deadline")
+                        ? new Deadline(nameDateTimeStrings.get(0), taskDate, taskTime)
+                        : new Event(nameDateTimeStrings.get(0), taskDate, taskTime));
+            } else {
+                return (taskType.equals("deadline") ? new Deadline(nameDateTimeStrings.get(0), taskDate)
```

Comment:
Perhaps the `? new Deadline(nameDateTimeStrings.get(0), taskDate)`  can be on a new line to maintain consistency and neatness.